src/command/restore/restore.c

```diff
@@ -1248,9 +1298,19 @@ restoreSelectiveExpression(Manifest *manifest)
             {
                 const String *db = strLstGet(dbList, dbIdx);
 
-                // Only user created databases can be zeroed, never system databases
-                if (cvtZToUInt64(strZ(db)) >= PG_USER_OBJECT_MIN_ID)
+                // Include system databases by default
+                if (strLstExists(excludeDbIdList, db) ||
+                    (strLstSize(includeList) > 0 && cvtZToUInt64(strZ(db)) >= PG_USER_OBJECT_MIN_ID))
                 {
+                    // Get db name to LOG
+                    for (unsigned int dbIdx = 0; dbIdx < manifestDbTotal(manifest); dbIdx++)
+                    {
+                        const ManifestDb *dbInfo = manifestDb(manifest, dbIdx);
+
+                        if (strEq(db, varStrForce(VARUINT(dbInfo->id))))
+                            LOG_DETAIL_FMT("database %s (%s) will be zeroed", strZ(db), strZ(dbInfo->name));
+                    }
```

Comment:
I'm a little worried about this FOR loop. If we have a million databases, we're just doing this to get the name to put into a detail-level log message. I know someone thought it might be nice, but I don't think we should take the hit. Instead, I think before the loop, we should output a `LOG_DETAIL_FMT("databases to be zeroed: %s", strZ(strLstJoin(dbList, ", ")));`
We always output the command and options provided to console if they have at least INFO set and always to the log file and they will (or should be) running the `pgbackrest info` command with the --set option on the backup they will use to restore if they are going to do selective restore and they can get the OIDs from that.

Reply:
I'm also in favor of using ids everywhere in the log details, for consistency reasons.
But as mentioned above, dbList would contain databases that don't have to be zeroed in case we use `--db-exclude` alone. So your log detail would probably not be accurate. But I'll remove the loop and the name ;-)